PocketCastsTests/Tests/IAP/IAPHelperTests.swift

```diff
@@ -48,40 +38,50 @@ final class IAPHelperTests: XCTestCase {
 
         let paymentFrequency = helper.getPaymentFrequency(for: .monthly)
         XCTAssertEqual(paymentFrequency, "month")
-
-        session.clearTransactions()
     }
 
     func testPurchase() throws {
-        let session = try SKTestSession(configurationFileNamed: configurationFile)
-        session.clearTransactions()
-        session.resetToDefaultState()
-        session.disableDialogs = true
-
-        let helper = IAPHelper()
-        SKPaymentQueue.default().add(helper)
-        defer {
-            SKPaymentQueue.default().remove(helper)
-        }
-        let expectation = XCTestExpectation(description: "Fetch Products")
-        NotificationCenter.default.addObserver(forName: ServerNotifications.iapProductsUpdated, object: nil, queue: nil) { notification in
-            expectation.fulfill()
-        }
         helper.requestProductInfo()
+        wait(for: ServerNotifications.iapProductsUpdated, timeout: iapTestTimeout, description: "Fetch Products")
 
-        wait(for: [expectation], timeout: iapTestTimeout)
         XCTAssert(helper.hasLoadedProducts)
 
-        let buyExpectation = XCTestExpectation(description: "Buy Product")
-        NotificationCenter.default.addObserver(forName: ServerNotifications.iapPurchaseCompleted, object: nil, queue: nil) { notification in
-            buyExpectation.fulfill()
-        }
         let buyResult = helper.buyProduct(identifier: .monthly)
         XCTAssert(buyResult)
 
-        wait(for: [buyExpectation], timeout: iapTestTimeout)
+        wait(for: ServerNotifications.iapPurchaseCompleted, timeout: iapTestTimeout, description: "Buy Products")
 
-        session.clearTransactions()
+        XCTAssertEqual(session.allTransactions().first?.productIdentifier, IAPProductID.monthly.rawValue)
```

Comment:
I added this check to verify the purchased product is the the correct one

Reply:
Good addition!